File: Hippo/Views/EpisodeCell.m
//
//  EpisodeCell.m
//  Hippo
//
//  Created by 전수열 on 13. 9. 7..
//  Copyright (c) 2013년 Joyfl. All rights reserved.
//

#import "EpisodeCell.h"

@implementation EpisodeCell

- (id)initWithReuseIdentifier:(NSString *)reuseIdentifier
{
	self = [super init];
	
	self.thumbnailView = [[UIImageView alloc] initWithFrame:CGRectMake( 0, 0, 71, 42 )];
	[self.contentView addSubview:self.thumbnailView];
	
	self.titleLabel = [[UILabel alloc] initWithPosition:CGPointMake( 80, 13 ) maxWidth:230];
	self.titleLabel.font = [UIFont systemFontOfSize:13];
	[self.contentView addSubview:self.titleLabel];
	
	self.bookmarkView = [[UIImageView alloc] initWithFrame:CGRectMake( 275, 0, 10, 16 )];
	self.bookmarkView.image = [UIImage imageNamed:@"icon_bookmark.png"];
	[self.contentView addSubview:self.bookmarkView];
	
	return self;
}

- (void)setEpisode:(Episode *)episode bookmark:(NSInteger)bookmark
{
	_episode = episode;
	_bookmark = bookmark;
	[self layoutContentView];
}

- (void)layoutContentView
{
	__block UIImageView *thumbnailView = self.thumbnailView;
	__block Episode *episode = self.episode;
	[self.thumbnailView setImageWithURL:[NSURL URLWithString:self.episode.thumbnail_url] placeholderImage:nil success:^(NSURLRequest *request, NSHTTPURLResponse *response, UIImage *image) {
		thumbnailView.image = episode.read.boolValue ? image.grayScaleImage : image;
		
	} failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error) {
		
	}];
	
	self.titleLabel.text = self.episode.title;
	self.titleLabel.textColor = episode.read.boolValue ? [UIColor grayColor] : [UIColor blackColor];
	[self.titleLabel sizeToFit];
	
	self.bookmarkView.hidden = self.episode.id.integerValue != self.bookmark;
}

@end

File: Hippo/Views/EpisodeCell.swift
//
//  EpisodeCell.swift
//  Hippo
//
//  Created by 전수열 on 10/6/14.
//  Copyright (c) 2014 Joyfl. All rights reserved.
//

import Foundation

class EpisodeCell: UITableViewCell {

    private var _episode: Episode?
    var episode: Episode? {
        get {
            return _episode
        }
        set {
            _episode = newValue
            let placeholder = UIImage(color: UIColor.whiteColor())
            self.thumbnailView.setImageWithURL(NSURL(string: newValue!.picture.url), placeholderImage: placeholder)
            self.titleLabel.text = newValue!.title
        }
    }

    let thumbnailView = UIImageView()
    let titleLabel = UILabel()
    let bookmarkView = UIImageView()

    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        self.contentView.addSubview(self.thumbnailView)
        self.contentView.addSubview(self.titleLabel)
        self.contentView.addSubview(self.bookmarkView)

        self.thumbnailView.snp_makeConstraints { make in
            make.width.equalTo(71)
            make.height.equalTo(42)
        }

        self.titleLabel.font = UIFont.systemFontOfSize(13)
        self.titleLabel.snp_makeConstraints { make in
            make.left.equalTo(80)
            make.top.equalTo(13)
            make.width.lessThanOrEqualTo(230)
        }

        self.bookmarkView.image = UIImage(named: "icon_bookmark")
        self.bookmarkView.snp_makeConstraints { make in
            make.left.equalTo(275)
            make.width.equalTo(10)
            make.height.equalTo(16)
        }
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    class var height: CGFloat {
        return 42
    }
}
